static/js/stream_edit.js

```diff
@@ -598,13 +598,17 @@ exports.initialize = function () {
             return;
         }
         var stream_name = stream_data.maybe_get_stream_name(stream_id);
-        var deactivate_stream_modal = templates.render("deactivation-stream-modal", {stream_name: stream_name});
-        $(".subscription_settings").append(deactivate_stream_modal);
+        var deactivate_stream_modal = templates.render("deactivation-stream-modal", {
+            stream_name: stream_name,
+            stream_id: stream_id,
+        });
+        $("#deactivation_stream_modal").remove();
+        $("#subscriptions_table").append(deactivate_stream_modal);
         overlays.open_modal('deactivation_stream_modal');
     });
 
     $("#subscriptions_table").on("click", "#do_deactivate_stream_button", function (e) {
-        var stream_id = get_stream_id(e.target);
+        var stream_id = $(e.target).data("stream-id");
```

Comment:
What's the story with this change?

Reply:
`get_stream_id` function fetch the id from `subscription_settings` element, but now this modal is no longer sub element or parent of `subscription_settings`. So we need to fetch id from modal itself.